app/hub/Views/UpgradePlanView/UpgradePlanView.jsx

```diff
@@ -0,0 +1,958 @@
+/**
+ * Upgrade Plan View Component
+ *
+ * Ghostery Browser Extension
+ * https://www.ghostery.com/
+ *
+ * Copyright 2019 Ghostery, Inc. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0
+ */
+
+import React, { useRef } from 'react';
+import ClassNames from 'classnames';
+import PropTypes from 'prop-types';
+import { NavLink } from 'react-router-dom';
+import globals from '../../../../src/classes/Globals';
+
+/**
+ * A React class component for rendering the Home View
+ * @return {JSX} JSX for rendering the Home View of the Hub app
+ * @memberof HubComponents
+ */
+const UpgradePlanView = (props) => {
+	const {
+		protection_level,
+		show_yearly_prices,
+		user,
+		actions,
+	} = props;
+
+	const {
+		toggleMonthlyYearlyPrices,
+		setBasicProtection,
+		setPlusProtection,
+		setPremiumProtection
+	} = actions;
+
+	const isPlus = (user && user.plusAccess) || false;
+	const isPremium = (user && user.premiumAccess) || false;
+
+	const BASIC = 'BASIC';
+	const PLUS = 'PLUS';
+	const PREMIUM = 'PREMIUM';
```

Comment:
Since these are not prop-dependent, we can move them outside the function into file scope and so avoid re-declaring them on each render.

Reply:
Good idea, changed to file scope!